src/DynamoCoreWpf/Utilities/NodeContextMenuBuilder.cs

```diff
@@ -0,0 +1,451 @@
+            //grid.Name = "mainTemplateGrid";
+            //grid.SetValue(WidthProperty, 220.0);
+            //grid.SetValue(HeightProperty, 30.0);
+            //grid.SetValue(MarginProperty, new Thickness(-15,0,0,0));
+            //grid.SetValue(HorizontalAlignmentProperty, HorizontalAlignment.Left);
+            //grid.SetValue(VerticalAlignmentProperty, VerticalAlignment.Stretch);
+            //grid.SetValue(BackgroundProperty, new SolidColorBrush(Colors.Transparent));
+
+            //FrameworkElementFactory textBlock = new FrameworkElementFactory(typeof(TextBlock));
+            //textBlock.SetValue(HorizontalAlignmentProperty, HorizontalAlignment.Stretch);
+            //textBlock.SetValue(MarginProperty, new Thickness(15, 0, 0, 0));
+            //textBlock.SetValue(IsHitTestVisibleProperty, false);
+            //textBlock.SetValue(VerticalAlignmentProperty, VerticalAlignment.Center);
+            //textBlock.SetValue(ForegroundProperty, new SolidColorBrush(Color.FromRgb(238,238,238)));
+
+            //dismissedAlertsMenuItem.ItemTemplate = itemTemplate;
+
+            //AddContextMenuItem(dismissedAlertsMenuItem, insertionPoint++);
+
+            return null;
+        }
+
+        private static void AddInjectedNodeViewCustomizationMenuItems(Dictionary<string, object> nodeViewCustomizationMenuItems)
+        {
+            foreach (KeyValuePair<string, object> keyValuePair in nodeViewCustomizationMenuItems)
+            {
+                ContextMenu.Items.Add(keyValuePair.Value);
+            }
+        }
+
+        private static MenuItem BuildIsInputMenuItem()
+        {
+            return CreateMenuItem
+            (
+                name: "isInput_cm",
+                header: Properties.Resources.NodeContextMenuIsInput,
+                isCheckable: true,
+                isChecked: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("IsSetAsInput"),
+                    Mode = BindingMode.TwoWay,
+                }
+            );
+        }
+
+        private static MenuItem BuildIsOutputMenuItem()
+        {
+            return CreateMenuItem
+            (
+                name: "isOutput_cm",
+                header: Properties.Resources.NodeContextMenuIsOutput,
+                isCheckable: true,
+                isChecked: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("IsSetAsOutput"),
+                    Mode = BindingMode.TwoWay,
+                }
+            );
+        }
+
+        private static MenuItem BuildHelpMenuItem()
+        {
+            return CreateMenuItem
+            (
+                name: "help_cm",
+                header: Properties.Resources.NodeContextMenuHelp,
+                command: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("ShowHelpCommand")
```

Comment:
can you use nameof?

Reply:
Have switched all PropertyPath("X") out for PropertyPath(nameof(NodeViewModel.X))